openquake/hazardlib/shakemap/conditioned_gmfs.py

```diff
@@ -0,0 +1,988 @@
+        RC = cov_WY_WD @ cov_WD_WD_inv
+
+        # Compute the scaling matrix "C" for the conditioned between-event
+        # covariance matrix
+        if native_data_available:
+            T_Y0 = tau_Y
+        else:
+            tau_zeros = numpy.zeros((num_target_sites, len(conditioning_imts)))
+            T_Y0 = numpy.block([tau_Y, tau_zeros])
+        C = T_Y0 - RC @ T_D
+
+        # Compute the conditioned within-event covariance matrix
+        # for the target sites
+        cov_WY_WY_wD = cov_WY_WY - RC @ cov_WD_WY
+
+        # Compute the "conditioned between-event" covariance matrix
+        # for the target sites
+        cov_BY_BY_yD = numpy.linalg.multi_dot([C, cov_HD_HD_yD, C.T])
+
+        # Finally, compute the conditioned mean
+        # of the ground motion at the target sites
+        mu_Y_yD = mu_Y + mu_BY_yD + RC @ (zeta_D - mu_BD_yD)
+
+        # And compute the conditional covariance
+        # of the ground motion at the target sites
+        cov_Y_Y_yD = cov_WY_WY_wD + cov_BY_BY_yD
+
+        # Store the results in a dictionary keyed by target IMT
+        # The four arrays below have different dimensions, so
+        # unlike the regular gsim get_mean_std, a numpy ndarray
+        # won't work well as the 4 components will be non-
+        # homogeneous
+        mu_Y_yD_dict[target_imt.string] = mu_Y_yD
+        cov_Y_Y_yD_dict[target_imt.string] = cov_Y_Y_yD
+        cov_WY_WY_wD_dict[target_imt.string] = cov_WY_WY_wD
+        cov_BY_BY_yD_dict[target_imt.string] = cov_BY_BY_yD
+
+    return mu_Y_yD_dict, cov_Y_Y_yD_dict, cov_WY_WY_wD_dict, cov_BY_BY_yD_dict
+
+
+def compute_spatial_cross_correlation_matrix(
+    sitecol_1,
+    sitecol_2,
+    imt_list_1,
+    imt_list_2,
+    spatial_correl,
+    cross_correl_within,
+):
+    # The correlation structure for IMs of differing types at differing locations
+    # can be reasonably assumed as Markovian in nature, and we assume here that
+    # the correlation between differing IMs at differing locations is simply the
+    # product of the cross correlation of IMs i and j at the same location
+    # and the spatial correlation due to the distance between sites m and n
+    # Can be refactored down the line to support direct spatial cross-correlation
+    # models
+    distance_matrix = geodetic_distance(
+        sitecol_1.lons.reshape(sitecol_1.lons.shape + (1,)),
+        sitecol_1.lats.reshape(sitecol_1.lats.shape + (1,)),
+        sitecol_2.lons,
+        sitecol_2.lats,
+    )
+    spatial_cross_correlation_matrix = numpy.block(
+        [
+            [
+                _compute_spatial_cross_correlation_matrix(
+                    distance_matrix,
+                    imt_1,
+                    imt_2,
+                    spatial_correl,
+                    cross_correl_within,
+                )
+                for imt_2 in imt_list_2
+            ]
+            for imt_1 in imt_list_1
+        ]
+    )
```

Comment:
We are not using black as code formatter, so no need to use its conventions.

Reply:
Reduced black usage